server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/IMAPServerModule.java

```diff
@@ -103,6 +106,10 @@ protected void configure() {
         Multibinder.newSetBinder(binder(), GuiceProbe.class).addBinding().to(ImapGuiceProbe.class);
 
         Multibinder.newSetBinder(binder(), CertificateReloadable.Factory.class).addBinding().to(IMAPServerFactory.class);
+
+        Multibinder.newSetBinder(binder(), ConnectionCheck.class)
+            .addBinding()
+            .to(AllowConnectionCheck.class);
```

Comment:
Just define the multibinder.

```suggestion
        Multibinder.newSetBinder(binder(), ConnectionCheck.class);
```

The binding to AllowConnectionCheck is useless as it is a Noop. Remove it.